probcomp/metaprob
# Add Docker image

## What does this do?
1. Adds files for building and manipulating a Docker image that, when run, starts Jupyter with all the necessary prerequisites for running the tutorial pre-installed.
2. Adds `make` targets for building and manipulating the above.
3. Modifies `.travis.yml` such that tests are run inside of a Docker container.
4. Fixes #60.

## Why should we do this?
As we've discovered recently installing Leiningen and other dependencies can be an error-prone process on some systems. Having a Docker image should make starting the tutorial painless for people who already have `docker` and `docker-compose` installed.

## How can I test this?

1. Install Docker on your system and ensure that `docker` is on your path.
2. Check out this branch.
3. `make docker-build`
4. `make docker-test`
5. Follow the instructions in the tutorial `README` to launch it.
6. Open the tutorial.
7. Try to run the first cell.
